contrib/runners/orquesta_runner/tests/unit/test_rerun.py

```diff
@@ -0,0 +1,172 @@
+    'publish_state',
+    mock.MagicMock(side_effect=mock_lv_ac_xport.MockLiveActionPublisher.publish_state))
+class OrquestRunnerTest(st2tests.WorkflowTestCase, st2tests.ExecutionDbTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(OrquestRunnerTest, cls).setUpClass()
+
+        # Register runners.
+        runnersregistrar.register_runners()
+
+        # Register test pack(s).
+        actions_registrar = actionsregistrar.ActionsRegistrar(
+            use_pack_cache=False,
+            fail_on_failure=True
+        )
+
+        for pack in PACKS:
+            actions_registrar.register_from_pack(pack)
+
+    @staticmethod
+    def get_runner_class(runner_name):
+        return runners.get_runner(runner_name).__class__
+
+    def test_rerun_option(self):
+        patched_orquesta_runner = self.get_runner_class('orquesta')
+
+        mock_rerun_result = (
+            ac_const.LIVEACTION_STATUS_RUNNING,
+            {'tasks': []},
+            {'execution_id': str(uuid.uuid4())}
+        )
+
+        with mock.patch.object(patched_orquesta_runner, 'rerun_workflow',
+                               mock.MagicMock(return_value=mock_rerun_result)):
+
+            wf_meta = self.get_wf_fixture_meta_data(TEST_PACK_PATH, TEST_FIXTURES['workflows'][0])
+            lv_ac_db1 = lv_db_models.LiveActionDB(action=wf_meta['name'], parameters=PARAM)
+            lv_ac_db1, ac_ex_db1 = ac_svc.request(lv_ac_db1)
+            self.assertFalse(patched_orquesta_runner.rerun_workflow.called)
+
+            # Rerun the execution.
+            context = {
+                're-run': {
+                    'ref': ac_ex_db1.id,
+                    'tasks': ['task1']
+                }
+            }
+
+            lv_ac_db2 = lv_db_models.LiveActionDB(action=wf_meta['name'], parameters=PARAM,
+                                                  context=context)
+            lv_ac_db2, ac_ex_db2 = ac_svc.request(lv_ac_db2)
+
+            self._wait_on_status(
+                lv_ac_db2,
+                ac_const.LIVEACTION_STATUS_RUNNING
+            )
+
+            options = {
+                'ref': ac_ex_db1.id,
+                'tasks': ['task1'],
+            }
+            patched_orquesta_runner.rerun_workflow.assert_called_with(ex_ref=ac_ex_db1,
+                                                                      options=options)
+
+    def test_rerun_with_invalid_workflow_id(self):
+        wf_meta = self.get_wf_fixture_meta_data(TEST_PACK_PATH, TEST_FIXTURES['workflows'][0])
+        lv_ac_db = lv_db_models.LiveActionDB(action=wf_meta['name'], parameters=PARAM)
+        lv_ac_db, ac_ex_db = ac_svc.create_request(lv_ac_db)
+
+        # Rerun option.
+        context = {
+            're-run': OPTIONS
+        }
+
+        orquesta_runner = self.get_runner_class('orquesta')
+        self.assertRaises(
+            Exception,
+            orquesta_runner.rerun_workflow,
+            ac_ex_db,
+            context
+        )
```

Comment:
Is there a way to know what the exception is? Of what the message? otherwise, this could be throwing some other exception and we will have no way of catching that.